CI `run-tests (8)` sometimes times out after 6 hours

From the CI run below, it might be that the JDK8 test runs has issues when the tasks `:iceberg-hive3:test` and `:iceberg-mr:test` run in parallel. Only JDK8 Gradle runs do include `:iceberg-hive3`, which explains why only the JDK8 CI test runs are affected (seen only the JDK8 test runs timing out).

I don't know about the inner workings of both project tests, but I suspect those two projects somehow interfere with each other.

If it's just those two test tasks, it might help to add a `mustRunAfter(':iceberg-mr:test')` to `:iceberg-hive3:test`.

[Example failute](https://github.com/apache/iceberg/pull/2450/checks?check_run_id=2335040662)
From the raw log (sorry, the raw-log expired, so can't link to it): (look at the timestamps)
```
2021-04-13T15:46:50.1563347Z Starting a Gradle Daemon (subsequent builds will be faster)
...
2021-04-13T15:51:44.2558196Z > Task :iceberg-hive3:test
...
2021-04-13T15:58:24.8564436Z > Task :iceberg-mr:test
...
2021-04-13T16:58:31.6625242Z > Task :iceberg-hive3:test
(reports test failure)
2021-04-13T20:50:20.3558417Z
(:iceberg-hive3:test reports next test failure)
...
2021-04-13T20:50:20.8582685Z > Task :iceberg-mr:test
(reports test failure)
2021-04-13T21:07:21.6565277Z > Task :iceberg-mr:test FAILED
...
2021-04-13T21:11:26.6561908Z > Task :iceberg-hive3:test FAILED
...
2021-04-13T21:11:26.6581909Z BUILD FAILED in 5h 24m 39s
```

--- a/.github/workflows/java-ci.yml
+++ b/.github/workflows/java-ci.yml
@@ -48,6 +48,7 @@ jobs:
         path: ~/.gradle/caches
         key: ${{ runner.os }}-gradle-${{ hashFiles('**/*.gradle') }}
         restore-keys: ${{ runner.os }}-gradle
+    - run: echo -e "$(ip addr show eth0 | grep "inet\b" | awk '{print $2}' | cut -d/ -f1)\t$(hostname -f) $(hostname -s)" | sudo tee -a /etc/hosts
     - run: ./gradlew check -Pquick=true -x javadoc
     - uses: actions/upload-artifact@v2
       if: failure()

